packages/plugin-ext/src/main/browser/command-registry-main.ts

```diff
@@ -60,6 +64,22 @@ export class CommandRegistryMainImpl implements CommandRegistryMain {
             return Promise.reject(e);
         }
     }
+
+    $getKeyBinding(commandId: string): PromiseLike<theia.CommandKeyBinding[] | undefined> {
+        try {
+            const keyBindings = this.keyBinding.getKeybindingsForCommand(commandId);
+            if (keyBindings) {
+                // transform inner type to CommandKeyBinding
+                return Promise.resolve(keyBindings.map(keyBinding => ({ id: commandId, value: keyBinding.keybinding })));
+            } else {
+                return Promise.resolve(undefined);
+            }
+
+        } catch (e) {
+            return Promise.reject(e);
```

Comment:
@benoitf just a code-style question, but what's the reasoning behind defining synchronous functions that return Promise objects at every return point, rather that just defining an async function and using regular return values and throwing errors? I don't really mind but I am still curious.

Reply:
@marechal-p copy/paste from other code of the class. Maybe it's PromiseLike vs Promise.

@evidolob ?

Reply:
Yes, it can be the case, maybe we should review our code

Reply:
ok, will create new issue around this